### PR TITLE
Core: log warning for unknown options

### PR DESCRIPTION
**Pull request by alwaysintreble**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
title

## How was this tested?
rolled seeds with valid and invalid option names

## If this makes graphical changes, please attach screenshots.
Screenshot_47

### CODE REVIEW

**Comment by KonoTyran** [comment]
So this will break things like mercenary mode yamls for alttp as they use and set "invalid" keys in the yaml.

**Comment by alwaysintreble** [comment]
It marks trigger option names as valid.

File: Generate.py [hunk 1]
```diff
@@ -8,9 +8,11 @@
 import urllib.parse
 import urllib.request
 from collections import Counter, ChainMap
+from itertools import chain
 from typing import Dict, Tuple, Callable, Any, Union
 
 import ModuleUpdate
+import worlds.AutoWorld
```
Comment (Berserker66) [inline Generate.py:15]: running this import before the update() call will crash with missing dependencies

Comment (alwaysintreble) [reply Generate.py:15]: ah PyCharm put it there automatically and since I didn't need to update didn't hit that. good to know.

**Comment by alwaysintreble** [comment]
Ran this against the weights file for the big async and turned out I didn't notice that `roll_triggers` was called 2 separate times and would give different results based on when it was called and would find invalid keys for other games even if that game wasn't being played. Fixed this by only doing the validation step after all triggers are finished resolving and having a global set of valid_keys to check against for both trigger resolutions now. Found a bunch of typos in the big async weights file with it and it succeeded with some custom names in it.

**Comment by Jarno458** [comment]
I am not sure if i like this, like for timespinner i can basicly use the same yaml for (that includes beta options) that works on both normal and beta. those new beta options simply wont have any effect but wont break anything either

**Comment by Jarno458** [comment]
but does it really matter? if the option is removed well it wont do a thing right? i dont think this warrents to error out

**Comment by Berserker66** [comment]
My proposal was to introduce hidden options, and therefore give maintainers the ability to make a hidden option that errors out when set, if they want to loudly deprecate an option. I too worry this will lead to more annoyance than it's worth, personally.

File: Generate.py [hunk 2]
```diff
@@ -487,6 +489,10 @@ def roll_settings(weights: dict, plando_options: PlandoOptions = PlandoOptions.b
 
     for option_key, option in world_type.options_dataclass.type_hints.items():
         handle_option(ret, game_weights, option_key, option, plando_options)
+    for option_key in game_weights:
+        if option_key in {"triggers", *valid_trigger_names}:
+            continue
+        logging.warning(f"{option_key} not a valid option name for {ret.game} and not present in triggers.")
```
Comment (Exempt-Medic) [inline Generate.py:495]:
```suggestion
        logging.warning(f"{option_key} is not a valid option name for {ret.game} and is not present in triggers.")
```

**Review by Exempt-Medic — APPROVED** [review]
Briefly looked at the code, merged into `main` and did functional testing, trying various methods of using triggers and options. It gave warnings only when it was supposed to. One small wording suggestion, otherwise it seems to work just fine

**Review by NewSoupVi — APPROVED** [review]
Since this is now a warning, I think I'm ok merging this. If there are issues because of a case we didn't consider, we can work them out and they at least won't break anything.

For the record, this really reaffirms my belief that the Webhost should output warnings somehow.

**Pull request merged**